src/target/target.h

```diff
@@ -66,6 +66,9 @@ enum target_state {
 	TARGET_HALTED = 2,
 	TARGET_RESET = 3,
 	TARGET_DEBUG_RUNNING = 4,
+	/* The target is unavailable for some reason. It might be powered down,
+	 * for instance. */
+	TARGET_UNAVAILABLE = 5
```

Comment:
Please, move the comment to the top of the enum (for consistency with the existing comments).

In what circumstances would you like to use this new target state? What prompted this change? Is it a part of soluton for a concrete issue?

Since the target_state enum is used globally throughout the whole OpenOCD, there may have unintended consequences. I am especially wary of code pieces that could be structured like this:

```
if (target->state != TARGET_HALTED) {
    /* Assume the target is running */

    /* Do something ... */
}
```

Reply:
Moved, and added a better commit message. If that doesn't answer your questions enough let me know and I'll expand on it there.

Reply:
Thank you.

From RISC-V point of view, this new state is perfectly fine. 

I'd only watch for existing OpenOCD code (outside of `riscv`) that uses `target->state` and that may not count with this newly added `TARGET_UNAVAILABLE` state. However, after quickly going through the existing code, the change looks safe.